### DISCUSSION TITLE
Workflows - Batched Content Releases

### DISCUSSION THREAD

**Discussion by Mosnar** [craftcms/cms#8941] [Ideas]
### Description

I came across this feature in [another CMS recently](https://user-guides.prismic.io/en/articles/778358-what-is-a-release) and thought it was brilliant and worth considering once workflows are on the table. The idea is that content changes can be staged to a specific release rather than published immediately. This would work really well in the Craft ecosystem where elements are extremely intertwined and highly related. The same concept is also used on Google Tag Manager, as another example. 

Here are a few example use-cases:
- Publishing a new series of products and related landing pages at the same time
- Publishing a blog post and product page at the same time
- Publishing a new category for a blog post at the same time as the blog post
- Elements created from within an inline-editor HUD could be marked as dependent on the entry for which it was created
- Rejected and unpublished content dependencies could safely be removed (think uploaded assets and call-to-action components)
- Opens the door for doing site history rollbacks across multiple elements

This becomes even more useful outside of workflows when you consider the static site generation implications - imagine not having to worry about triggering 14 redundant builds while adding a blog post due to a bunch of elements being saved during the editing process.

This is practically begging to be done now that the work adding delta saves/change histories, and element revisions is done!

**Comment by Kethatril** [comment]
SilverStripe is another CMS that has this feature (https://userhelp.silverstripe.org/en/4/creating_pages_and_content/campaigns/)

**Comment by RitterKnightCreative** [comment]
Agree wholeheartedly with this. Similar to my feature request a few years ago: https://github.com/craftcms/cms/issues/2116.

**Comment by brandonkelly** [comment]
Actually have been thinking about this the past few days. Biggest thing holding me back is the naming. “Release” and “campaign” have both been considered but not a huge fan of either of them.

**Comment by Mosnar** [comment]
Agreed, not a fan of those names. Here are a few more ideas:
- Project
- Changeset
- Branch
- Workspace
- Batch
- Plan
- Stage

**Comment by mmikkel** [comment]
I don't have an actual name in mind for this feature, but the way I _think_ about it is that it'd be sort of like a "site draft", in that it could basically enable you to put the whole site (or _sites_) in sort of a draft mode, where authors could add, remove or change content in multiple sections, preview (and share) the whole site as it will look when all of those content changes are made live, and finally, be able to either discard or publish all those changes at once. "Site Draft" or "Site Version" is kinda problematic, but yeah... that's how I think about the concept, maybe a better name could be derived from that line of thinking.

**Comment by brandonkelly** [comment]
Batch is pretty good.

**Comment by AugustMiller** [comment]
Logging another platform that has implemented something like this:
https://graphcms.com/docs/guides/content/scheduled-publishing

**Comment by sjcallender** [comment]
This is likely easier in Craft 4 given all the performance improvements around duplicating content for drafts and versions. We have a site where we've built this feature for them and the slowness when publishing all the specific drafts on the predetermined date is the only pain point currently.
